src/devices/Lm75/Lm75.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Lm75
+{
+    /// <summary>
+    /// Digital Temperature Sensor LM75
+    /// </summary>
+    public class Lm75 : IDisposable
+    {
+        private I2cDevice _sensor;
+
+        /// <summary>
+        /// LM75 I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x48;
+
+        #region prop
+
+        /// <summary>
+        /// LM75 Temperature (℃)
+        /// </summary>
+        public double Temperature { get => GetTemperature(); }
+
+        private bool _disable;
+        /// <summary>
+        /// Disable LM75
+        /// </summary>
+        public bool Disabled { get { return _disable; } set { SetShutdown(value); _disable = value; } }
+
+        #endregion
+
+        /// <summary>
+        /// Creates a new instance of the LM75
+        /// </summary>
+        /// <param name="sensor">I2C Device, like UnixI2cDevice or Windows10I2cDevice</param>
+        public Lm75(I2cDevice sensor)
+        {
+            _sensor = sensor;
+
+            Disabled = false;
+        }
+
+        /// <summary>
+        /// Read LM75 Temperature (℃)
+        /// </summary>
+        /// <returns>Temperature</returns>
+        private double GetTemperature()
+        {
+            Span<byte> readBuff = stackalloc byte[2];
+
+            _sensor.WriteByte((byte)Register.LM_TEMP);
+            _sensor.Read(readBuff);
+
+            // Details in Datasheet P10
+            double temp = 0;
+            ushort raw = (ushort)((readBuff[0] << 3) | (readBuff[1] >> 5));
+            if ((readBuff[0] & 0x80) == 0)
+            {
+                temp = raw * 0.125;
+            }
+            else
+            {
+                // two's complement
+                raw |= 0xF800;
```

Comment:
isn't two's complement be equivalent to using BinaryPrimitives.ReadLittle/BigEndian?

Reply:
Aha, I don't quite understand what you mean. `using BinaryPrimitives.ReadInt16Little/BigEndian` need to pass in a Span<T>. This is a binary complement for negative numbers. Temperature register data contains undefined bits, and BinaryPrimitives.ReadInt16Little/BigEndian cannot be used directly😄. @krwq

Reply:
@ZhangGaoxing I'm fine with this as is for now but I suspect if you just cast raw to short you might get same result - not sure what the `raw |= 0xF800` does though - I'd have to dig more to check it so fine to leave as is if you get correct results

Reply:
`0xF800` just sets the high five unused bits